spec/cspec/c/gen-config-thy.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021, Proofcraft Pty Ltd
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+
+#
+# This script generates a theory Kernel_Config.thy from the file gen_config.h,
+# which in turn is produced by the kernel build system, containing kernel
+# configuration options.
+#
+# The script makes the following assumptions about gen_config.h:
+#
+#   - only lines that start with "#define" are config options that are used
+#   - the value of the config option is separated from its name by two spaces
+#   - there is an optional comment after the value, separated by two spaces
+#   - values are numbers, booleans or strings
+#
+
+import re
+
+from datetime import date
+from os import path, getenv
+from typing import Dict
+
+# Isabelle types for config keys
+bool = 'boolean'
+string = 'string'
+nat = 'nat'
+word = 'machine_word'
+
+# All known config keys that could be set in a gen_config.h file.
+# The dict maps config key to (type, definition name)
+#
+# This table duplicates some information from the kernel build system, but
+# only which keys exist, not their values.
```

Comment:
It will by default ignore new keys. The idea is that the proofs will fail if the value was used in a verification-visible way in the C, but should keep succeeding when not. That way this won't block things if a new debug setting is added for instance, but will fail the regression test if the verification-relevant values are refactored somehow.

Reply:
Perfect, that's even better! Thanks for the clarification.